app/src/main/java/org/mozilla/focus/tabs/MergeTabsMiddleware.kt

```diff
@@ -30,7 +31,9 @@ class MergeTabsMiddleware(
         action: BrowserAction
     ) {
         val multiTabsFeature = FocusNimbus.features.tabs
-        if (multiTabsFeature.value(appContext).isMultiTab || action !is TabListAction.AddTabAction) {
+        if ((multiTabsFeature.value(appContext).isMultiTab && !appContext.components.settings.shouldMergeTopSiteTab) ||
```

Comment:
This middleware was intended to be used for the `multiTabsFeature` experiment, so I do not recommend changing the logic here. I filed https://github.com/mozilla-mobile/focus-android/issues/7616 to remove this code instead.

See my other comment about what we could do to fix this bug.